static/js/entry/zendesk.js

```diff
@@ -0,0 +1,172 @@
+  // are called at the appropriate time. We have to ensure that the iframe
+  // exists before we can start polling to see if the content has loaded
+  // (so that we can call functions to fiddle with the HTML and styling inside
+  // of it).
+
+  launcherLoaded: () => {
+    const iframe = document.querySelector("iframe.zEWidget-launcher")
+    if (_.isNull(iframe)) {
+      return
+    }
+
+    const btn = iframe.contentDocument.querySelector(".u-userLauncherColor")
+    if (_.isNull(btn)) {
+      return
+    }
+
+    const regularBackgroundColor = "rgba(0, 0, 0, .14)"
+    const defaultHoverBackgroundColor = "#a31f34" // fall back color
+    const hoverBackgroundColor = window.getComputedStyle(
+      btn,
+      defaultHoverBackgroundColor
+    ).backgroundColor
+    // We need to set a new background color, and unfortunately,
+    // the existing background color is set with "!important".
+    // As a result, the only way to override this existing color is to
+    // *also* use "!important".
+    const setHover = () => {
+      btn.style.setProperty(
+        "background-color",
+        hoverBackgroundColor,
+        "important"
+      )
+    }
+    const unsetHover = () => {
+      btn.style.setProperty(
+        "background-color",
+        regularBackgroundColor,
+        "important"
+      )
+    }
+    btn.onmouseenter = setHover
+    btn.onmouseleave = unsetHover
+    unsetHover()
+  },
+  webWidgetLoaded: () => {
+    const iframe = document.querySelector("iframe.zEWidget-webWidget")
+
+    // this is Zendesk's ID for the program selector field
+    // which we want to hide in Discussions
+    const programFieldName = "24690866"
+    const programEl = iframe.contentDocument.querySelector(
+      `input[name="${programFieldName}"]`
+    )
+
+    if (programEl) {
+      programEl.parentNode.parentNode.style.setProperty(
+        "display",
+        "none",
+        "important"
+      )
+    }
+  }
+}
+
+const zendeskPollForExistence = name => {
+  let tries = 0
+  const intervalID = setInterval(() => {
+    tries += 1
+    const iframe = document.querySelector(`iframe.zEWidget-${name}`)
+    if (iframe) {
+      clearInterval(intervalID)
+      zendeskPollForLoaded(name)
+    } else if (tries > 100) {
+      // max 100 tries (10 seconds)
+      console.error(`couldn't find Zendesk iframe: ${name}`) // eslint-disable-line no-console
+      clearInterval(intervalID)
+    }
+  }, 100) // check every 100 milliseconds
+}
+
+const zendeskPollForLoaded = name => {
+  let tries = 0
+  let iframeDocument = null
+  const intervalID = setInterval(() => {
+    tries += 1
+    const iframe = document.querySelector(`iframe.zEWidget-${name}`)
+    try {
+      iframeDocument = iframe.contentDocument
+    } catch (err) {
+      // cross-domain exception: can't continue
+    }
+    if (!iframeDocument) {
+      console.error(`Can't access content of Zendesk iframe: ${name}`) // eslint-disable-line no-console
```

Comment:
Did we want to remove this because we concluded it wasn't useful?

Reply:
The problem before was that we were using this function (`zendeskPollForLoaded`) to check for the existence of a few iframes which the Zendesk embed doesn't make any longer (they were called `ticketSubmissionForm`, `nps`, and `imp`). I changed it so we're just checking for the ones it actually creates.

Reply:
Also that this block is causing `webWidgetLoaded` to sometimes not be called.